semmc-ppc/src/SemMC/Architecture/PPC32.hs

```diff
@@ -363,56 +364,56 @@ operandValue sym locLookup op = TaggedExpr <$> operandValue' op
           let loc = LocGPR (PPC.GPR gpr)
           in if gpr /= 0
              then A.LocationOperand loc <$> locLookup loc
-             else A.ValueOperand <$> S.bvLit sym knownNat 0
-        operandValue' (PPC.I1imm (I.I x)) = A.ValueOperand <$> S.bvLit sym knownNat (toInteger x)
-        operandValue' (PPC.I32imm (I.I x)) = A.ValueOperand <$> S.bvLit sym knownNat (toInteger x)
+             else A.ValueOperand <$> S.bvLit sym knownNat (BV.zero knownNat)
+        operandValue' (PPC.I1imm (I.I x)) = A.ValueOperand <$> S.bvLit sym knownNat (BV.mkBV knownNat (toInteger x))
```

Comment:
Is `toInteger` here safe/lossless?  I know there are safe extraction functions from the `I` type, but this looks like it is pulling out the underlying integer directly?

Reply:
Not sure -- I was only mimicking what was there. At the very least, this shouldn't be breaking anything that wasn't already broken, since the old version was doing the exact same thing.